core/embed/rust/src/ui/layout/obj.rs

```diff
@@ -273,7 +273,7 @@ impl LayoutObj {
         static TYPE: Type = obj_type! {
             name: Qstr::MP_QSTR_Layout,
             locals: &obj_dict!(obj_map! {
-                Qstr::MP_QSTR_set_timer_fn => obj_fn_2!(ui_layout_set_timer_fn).as_obj(),
+                Qstr::MP_QSTR_attach_timer_fn => obj_fn_2!(ui_layout_attach_timer_fn).as_obj(),
```

Comment:
this makes it sound like we are "attaching a timer function", but instead this is "set timer function and attach to tree", right?

maybe we could call this `init` ?

Reply:
That's correct. `init` sounds rather nebulous, what about something like `init_with_timer_fn` or `set_timer_fn_and_attach`?

Reply:
Ah, right, I understand the issue a little better now.

We need to send the `Attach` event at the time when the layout actually goes to screen. (this is now `handle_input_and_rendering`). The "set timer callback" operation is a functionally separate thing, but because we can't get the timer in constructor (where it seems to belong), we bundle the two things together.

("set timer callback" can be called arbitrarily and the implementation does not care for as long as it still receives the timer events when expected. "attach" can be called repeatedly to indicate that the same layout instance is put on screen again)

(speaking of, the current `Timer` implementation is very fragile. it doesn't clean up after itself, and if `handle_timers` did any await other than on that one timer, it would mess things up because it happily schedules the task multiple times. we should clean that up before this goes to production.
in fact a good opportunity would be allowing a single global layout with a single global timer that would reset all scheduled events every time something went on screen. this thing might then as well live mostly in Rust.)

considering the parenthesized content :) let's keep the name as-is and make it better in the future